pkg/resource/deploy/step.go

```diff
@@ -948,32 +938,51 @@ func (s *ImportStep) Apply(preview bool) (resource.Status, StepCompleteFunc, err
 		}
 
 		if len(inputProperties) == 0 {
-			logging.V(9).Infof("Importing %v with required properties", s.URN())
-
-			for _, p := range r.InputProperties {
-				if p.IsRequired() {
-					k := resource.PropertyKey(p.Name)
-					s.new.Inputs[k] = s.old.Inputs[k]
-				}
-			}
+			logging.V(9).Infof("Importing %v with all properties", s.URN())
+			s.new.Inputs = s.old.Inputs.Copy()
 		} else {
 			logging.V(9).Infof("Importing %v with supplied properties: %v", s.URN(), inputProperties)
-
 			for _, p := range inputProperties {
 				k := resource.PropertyKey(p)
 				if value, has := s.old.Inputs[k]; has {
 					s.new.Inputs[k] = value
 				}
 			}
 		}
```

Comment:
Should we change the message below if these inputs fail `Check`? These are user-provided, so it seems that a validation failure is probably not a provider bug.

Reply:
Yes, I think that makes sense.